client/src/main/java/io/pravega/client/tables/IteratorArgs.java

```diff
@@ -38,21 +42,76 @@ static IteratorArgs fromBytes(ByteBuffer buffer) {
         throw new UnsupportedOperationException("PrimaryKey range iterators not supported.");
     }
 
+    /**
+     * Creates a new {@link IteratorArgs} instance for an iterator that returns {@link TableKey}/{@link TableEntry}
+     * instances with the same Primary Key (see {@link TableKey#getPrimaryKey()} and all Secondary keys (see
+     * {@link TableKey#getSecondaryKey()}) that begin with the given prefix. Depending on the arguments provided, this
+     * may iterate through all {@link TableKey}/{@link TableEntry} instances matching the given Primary Key or a sub-range.
+     *
+     * @param primaryKey         A {@link ByteBuffer} representing the full Primary Key to search.
+     * @param secondaryKeyPrefix A {@link ByteBuffer} that indicates the Secondary Key prefix for all {@link TableKey}/
+     *                           {@link TableEntry} instances to return. If not provided (null), the iterator will list
+     *                           all {@link TableKey}/{@link TableEntry} instances that match the given Primary Key.
+     * @return A new {@link IteratorArgs} that can be passed to {@link KeyValueTable#keyIterator} or to
+     * {@link KeyValueTable#entryIterator}.
+     */
     static IteratorArgs forPrimaryKey(@NonNull ByteBuffer primaryKey, @Nullable ByteBuffer secondaryKeyPrefix) {
         throw new UnsupportedOperationException("PrimaryKey prefix iterators not supported.");
     }
 
-    // Range iterator between 2 PKs (Optional)
+    /**
+     * Creates a new {@link IteratorArgs} instance for an iterator that returns {@link TableKey}/{@link TableEntry}
+     * instances with Primary Keys (see {@link TableKey#getPrimaryKey()} between the two values. Depending on the arguments
+     * provided, this may iterate through all {@link TableKey}/{@link TableEntry} instances within the {@link KeyValueTable}
+     * or a sub-range.
+     *
+     * @param fromPrimaryKey A {@link ByteBuffer} that indicates the Primary Key (see {@link TableKey#getPrimaryKey()})
+     *                       to begin the iteration at. If not provided (null), the iterator will begin with the first
+     *                       {@link TableKey}/{@link TableEntry} in the {@link KeyValueTable}. This argument is inclusive.
+     * @param toPrimaryKey   A {@link ByteBuffer} that indicates the Primary Key (see {@link TableKey#getPrimaryKey()})
+     *                       to end the iteration at. If not provided (null), the iterator will end with the last
+     *                       {@link TableKey}/{@link TableEntry} in the {@link KeyValueTable}. This argument is inclusive.
+     * @return A new {@link IteratorArgs} that can be passed to {@link KeyValueTable#keyIterator} or to
+     * {@link KeyValueTable#entryIterator}.
+     */
     static IteratorArgs forRange(@Nullable ByteBuffer fromPrimaryKey, @Nullable ByteBuffer toPrimaryKey) {
```

Comment:
I am surprised we would support this because we hash the primary keys to obtain the segment, so this can't operate against a subset of the segments even when the range is small. Also even though it's a iterator, this can't possibly do this in order right?

Reply:
When implemented (not necessarily as part of this initial revamp (release 0.10), this will be a distributed iterator across all partitions. It will issue the request to all table partitions and merge the result on the client-side. Since all Keys are stored in order in each Table Segment, this will be a "simple" merge iterator. The difficulty will be from issuing all those parallel requests and managing them - that's why I am willing to push this to a subsequent update (it's defined in the API, but may be implemented after 0.10). 

This was described in the PDP and #5931 (see last task at the bottom about global iterators).